cpp/ql/src/Security/CWE/CWE-311/CleartextBufferWrite.ql

```diff
@@ -13,21 +13,32 @@
 
 import cpp
 import semmle.code.cpp.security.BufferWrite
-import semmle.code.cpp.security.TaintTracking
 import semmle.code.cpp.security.SensitiveExprs
-import TaintedWithPath
+import semmle.code.cpp.security.Security
+import semmle.code.cpp.dataflow.TaintTracking
```

Comment:
This is going from IR taint tracking to AST taint tracking, which is a bit sad. Could we replace this with `semmle.code.cpp.ir.dataflow.TaintTracking`?

Reply:
Ah that's interesting, I thought the the security lib / default taint tracking used AST taint.

I'll update it to be IR-based...

Reply:
Fantastic! Here's a quick summary:
- `semmle.code.cpp.security.TaintTracking` is an alias for `semmle.code.cpp.ir.dataflow.DefaultTaintTracking` (which is an IR taint tracking library that uses an IR `DataFlow` configuration)
- `semmle.code.cpp.dataflow.TaintTracking` is an AST taint tracking library
- `semmle.code.cpp.ir.dataflow.TaintTracking` is an IR taint tracking library built on an IR `TaintTracking` configuration. By now I think this library should mostly be superior to `semmle.code.cpp.dataflow.TaintTracking`, but there are a few limitations that mostly shouldn't matter in practice.

Reply:
I've just done two things:
 - put it back on IR taint tracking, but no longer via relying on DefaultTaintTracking.  This is probably slightly untidier than what we started with, but opens the possibility to replace security library `isUserInput` with modern flow sources (which I should look at later today...)
 - I've also added to QLDoc at the top of two of the library files you described above.  That should help us (and others) remember which is which in future.

Reply:
I've updated it to use `FlowSource`, which is cleaner and appears to give better textual explanations of the source.